### PR TITLE
Test different branches of django-ex on different platforms

### PR DESCRIPTION
**Pull request by frenzymadness**
We need to test the master branch of django-ex on RHEL/Centos 7 and the 2.2.x branch on RHEL 8 and Fedora.

Depends on this change in common scripts: https://github.com/sclorg/container-common-scripts/pull/187

### CODE REVIEW

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
Common scripts are broken: https://github.com/sclorg/container-common-scripts/issues/190 :disappointed:

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
One more condition needed, Django 2 has no support for Python 2.

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
This seems to be ready for the final review. centos7 - openshift fails on too many requests to Docker hub and Jenkins-CI for rhel7 - openshift fails on some openshift stuff (cc @phracek).

I've verified that it tests the proper version of Django (from django-ex project) on the proper platform and python version. Cc @torsava

**Comment by phracek** [comment]
I don't follow the problem with OpenShift. It looks like OpenShift 3 does not start up properly.
Let's try it one more time.

[test-openshift]

**Comment by phracek** [comment]
RHEL7 OpenShift now failed on DockerHub rate limit too.

What about changing https://github.com/sclorg/s2i-python-container/blob/master/test/run-openshift#L49 from postgresql-9.6 to postgresql-10 and we can use an image from Quay.io?
postgresql-9.6 is not supported though.

**Comment by frenzymadness** [comment]
> RHEL7 OpenShift now failed on DockerHub rate limit too.
> 
> What about changing https://github.com/sclorg/s2i-python-container/blob/master/test/run-openshift#L49 from postgresql-9.6 to postgresql-10 and we can use an image from Quay.io?
> postgresql-9.6 is not supported though.

Good idea. Could you please prepare a pull request? I can then rebase this pull request and tests should be all right.

File: test/run [hunk 1]
```diff
@@ -216,13 +216,25 @@ test_application() {
 
 test_from_dockerfile(){
   info "Test from Dockerfile"
+  # Due to the old SQLite, the latest LTS release of Django
+  # does not work on RHEL 7. We are testing old Django example
+  # from master branch on EL 7 and the new one from 2.2.x branch
+  # on all other platforms. Also, Django 2 does not support Python 2
+  # so we have to test the older version on newer platforms (EL 8)
+  # with older Python.
```
Comment (torsava) [inline test/run:224]: I gotta admit I was a bit confused here, due to the new/old Python, new/old Django, new/old Platform. How about:

> The latest LTS release of Django does not work on RHEL 7 (due to an old version of SQLite), and no longer supports Python 2. So on all CentOS/RHEL 7 images and all Python 2 images, we're using the old Django version 2.2.x. In all other cases we're using the latest LTS release of Django from the main branch of the django-ex project.

**Comment by torsava** [comment]
I opened one comment, besides that, LGTM!

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
[test][test-openshift]

**Comment by frenzymadness** [comment]
@phracek could you please take a look here on the CI?

**Comment by phracek** [comment]
@frenzymadness It looks like it depends on https://github.com/sclorg/s2i-python-container/pull/429

On RHEL-7 DockerHub rate limit.

```
rying to pull repository registry.access.redhat.com/centos/postgresql-96-centos7 ... 
Pulling repository registry.access.redhat.com/centos/postgresql-96-centos7
Trying to pull repository registry.redhat.io/centos/postgresql-96-centos7 ... 
Trying to pull repository docker.io/centos/postgresql-96-centos7 ... 
Trying to pull repository docker.io/centos/postgresql-96-centos7 ... 
toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
NAME                                    DOCKER REPO                                TAGS                              UPDATED
imagestream.image.openshift.io/python   172.30.1.1:5000/sclorg-test-10503/python   2.7,3.8-ubi8,latest + 5 more...   6 seconds ago
In project sclorg-test-10503 on server https://127.0.0.1:8443
You have no services, deployment configs, or build configs.
Run 'oc new-app' to create an application.
```

On CentOS CI there is another problem
```
14:27:04 Error from server (Forbidden): pods is forbidden: User "developer" cannot list pods in the namespace "sclorg-test-14591": no RBAC policy matched
14:27:04 Error from server (Forbidden): replicationcontrollers is forbidden: User "developer" cannot list replicationcontrollers in the namespace "sclorg-test-14591": no RBAC policy matched
14:27:04 Error from server (Forbidden): services is forbidden: User "developer" cannot list services in the namespace "sclorg-test-14591": no RBAC policy matched
14:27:04 Error from server (Forbidden): daemonsets.apps is forbidden: User "developer" cannot list daemonsets.apps in the namespace "sclorg-test-14591": no RBAC policy matched
14:27:04 Error from server (Forbidden): deployments.apps is forbidden: User "developer" cannot list deployments.apps in the namespace "sclorg-test-14591": no RBAC policy matched
14:27:04 Error from server (Forbidden): replicasets.apps is forbidden: User "developer" cannot list replicasets.apps in the namespace "sclorg-test-14591": no RBAC policy matched

```

It looks like when the tests are finished and cluster is going to shutting down, then it is used user `developer` instead of `system:admin`.
@vrutkovs @gabemontero Can OpenShift user 'developer' delete pods,pvc,services, etc? And even shutting down cluster?

**Comment by frenzymadness** [comment]
Rebased, [test][test-openshift]

**Comment by phracek** [comment]
CentOS7 failed on command-app which pulls `ubi7/ubi` image https://github.com/sclorg/container-common-scripts/blob/master/test-lib-openshift.sh#L1000.

```
12:17:54 python-27-centos7-testing-2-build   0/1       Completed          0          3m
12:17:57 .NAME                                READY     STATUS             RESTARTS   AGE
12:17:58 command-app                         0/1       ImagePullBackOff   0          2m
12:17:58 python-27-centos7-testing-1-5d4s7   1/1       Running            0          3m
12:17:58 python-27-centos7-testing-1-build   0/1       Error              0          4m
12:17:58 python-27-centos7-testing-2-build   0/1       Completed          0          3m
12:18:02 .FAIL
12:18:02 error: unable to upgrade connection: container not found ("command-container")
```

**Review by phracek — APPROVED** [review]
Thanks for removing `.travis.yaml` file. 

It looks good to me.

**Pull request merged**